Source/WebCore/dom/Node.cpp

```diff
@@ -2079,8 +2079,8 @@ static void showSubTreeAcrossFrame(const Node* node, const Node* markedNode, con
     fputs(indent.utf8().data(), stderr);
     node->showNode();
     if (!node->isShadowRoot()) {
-        if (node->isFrameOwnerElement())
-            showSubTreeAcrossFrame(static_cast<const HTMLFrameOwnerElement*>(node)->contentDocument(), markedNode, indent + "\t");
+        if (auto* frameOwner = dynamicDowncast<HTMLFrameOwnerElement>(node))
+            showSubTreeAcrossFrame(frameOwner->contentDocument(), markedNode, indent + "\t");
```

Comment:
I could see that we might want to use protectedContentDocument() here. If #23506 looks good and can land I can rebase this on top of that and make that change.

Reply:
Yes, let's leverage `protectedContentDocument()` here.